pages/tutorials/graphql_sso.md.erb

```diff
@@ -0,0 +1,289 @@
+
+- to set up your organisation for SSO, you need your internal Buildkite id
+- use the get method with your org slug to retrieve your org id
+- this id will be required in all other sso methods
+
+For example, the following code will retrieve any SSO providers that have been created for the organisation `Acme-Inc`:
+
+```graphql
+query GetProviders {
+  organization(slug:"acme-inc"){
+    id
+    ssoProviders(first: 50) {
+      edges {
+        node {
+          id
+          url
+        }
+      }
+    }
+  }
+}
+```
+
+### Create your SSO provider
+
+- use the create method to create a blank 
+- after you set up your id provider, you will come back and fill in the provider details using the update method
+- use the org id you retreived from the GET method 
+
+The following example code creates a blank SSO provider:
+
+```graphql
+mutation MakeBlankProvider {
+  ssoProviderCreate(input:{
+    organizationId:"<paste your organization id here>",
+    type:SAML
+  }) {
+    ssoProvider {
+      id
+      type
+      ...on SSOProviderSAML {
+        serviceProvider {
+          issuer
+          ssoURL
+          metadata {
+            url
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+- can choose from three different provider types: `SAML`, `GOOGLE_OAUTH`, and `GITHUB_OAUTH`
+
+SAML:
+- if you haven't set up your identity provider with buildkite, do that before continuing
+- if you've already set up your identity provider, see [the examples below](#create-a-saml-provider-with-data) for code to create your identity provider with data
+
+Google_oauth:
+- add the two required fields for google authentication:
+    - googleHostedDomain: the domain your team uses with Google G Suite as a string
+    - discloseGoogleHostedDomain: a boolean describing whether or not you'd like your domain name to be listed on the authentication page 
+
+```graphql
+mutation CreateGoogleProvider {
+  ssoProviderCreate(input:{
+    organizationId:"<paste your organization id here>",
+    type:GOOGLE_OAUTH,
+    googleHostedDomain:"coffee.com",
+    discloseGoogleHostedDomain: false
+  }) {
+    ssoProvider {
+      id
+      url
+      ...on SSOProviderGoogleOAuth {
+        googleHostedDomain
+        discloseGoogleHostedDomain
+      }
+    }
+  }
+}
+```
+
+github_oauth:
+- add your github organisation name:
+    - githuborganisationname: 'test'
+
+```graphql
+mutation CreateGitHubProvider {
+  ssoProviderCreate(input:{
+    organizationId:"<paste your organization id here>",
+    type:GITHUB_OAUTH,
+    githubOrganizationName:"buildkite"
+  }) {
+    ssoProvider {
+      id
+      url
+      ...on SSOProviderGithubOAuth {
+        githubOrganizationName
+      }
+    }
+  }
+}
+```
+
+- test the login flow, ensure that a user can successfully complete the new authentication and gain access to your buildkite organisation before continuing
+- the enable method will error if a successful test hasn't occurred
+
+### Enable the provider for your organisation
```

Comment:
Would be good to mention that Enabling SSO marks all current users as requiring SSO. The user will likely want to mark external staff and robot users as not needing SSO manually after this.

Reply:
Oh cool, how does one go about marking people as not needing SSO? I thought that it was an all-or-nothing kind of thing.